osu.Game/Overlays/Comments/CommentsContainer.cs

```diff
@@ -117,39 +136,69 @@ public CommentsContainer(CommentableType type, long id)
         private void load()
         {
             background.Colour = colours.Gray2;
+            placeholderBackground.Colour = colours.Gray3;
         }
 
         protected override void LoadComplete()
         {
-            Sort.BindValueChanged(onSortChanged, true);
+            Sort.BindValueChanged(_ => updateComments());
             base.LoadComplete();
         }
 
-        private void onSortChanged(ValueChangedEvent<CommentsSortCriteria> sort)
+        public void ShowComments(CommentableType type, long id)
+        {
+            parameters = new CommentBundleParameters(type, id);
+            updateComments();
+        }
+
+        public void ShowComments(CommentBundle commentBundle)
         {
+            parameters = null;
             clearComments();
-            getComments();
+            onSuccess(commentBundle);
         }
 
-        private void getComments()
+        private void updateComments()
         {
-            request?.Cancel();
-            loadCancellation?.Cancel();
-            request = new GetCommentsRequest(type, id, Sort.Value, currentPage++);
-            request.Success += onSuccess;
-            api.Queue(request);
+            if (parameters == null)
+                return;
+
+            clearComments();
+            getComments();
         }
 
         private void clearComments()
         {
+            request?.Cancel();
+            loadCancellation?.Cancel();
             currentPage = 1;
             deletedChildrenPlaceholder.DeletedCount.Value = 0;
-            moreButton.IsLoading = true;
+            noCommentsPlaceholder.Hide();
             content.Clear();
+            moreButton.IsLoading = true;
+            moreButton.Show();
+        }
+
+        private void getComments()
+        {
+            if (parameters == null)
+                return;
+
+            request = new GetCommentsRequest(parameters, Sort.Value, currentPage++);
+            request.Success += onSuccess;
+            api.Queue(request);
         }
 
         private void onSuccess(CommentBundle response)
```

Comment:
Maybe not necessarily relevant right here, but this handler is getting really big and its body is hard to parse. Would be useful to split it into smaller, more well-defined methods, to be able to tell quicker what is going on.